### DISCUSSION TITLE
Gestione voti

### DISCUSSION THREAD

**Discussion by Faobi69** [iisgiua/giuaschool#114] [Idee e proposte]
Inoltre avrei tre richieste sulle valutazioni se è possibile:

 -   Possibilità di impostare le valutazioni numeriche quindi scegliere se sono abilitati i voti col + o -, solo col mezzo voto e solo interi. (io per esempio non vorrei che ci fossero i + e i - ma solo voti interi o mezzi)
-    Possibilità di scegliere il voto minimo e voto massimo assegnabile
-    Possibilità di scegliere un numero massimo di voti inseribili per tiplogia per esempio 4+4+4 per periodo scolastico
-    Possibilità da parte dei docenti di inserire il voto in qualsiasi giornata loro vogliano senza il vincolo di aver firmato una lezione in quella giornata per quella materia

**Comment by trinko** [comment]
>     * Quando si inseriscono gli alunni non si possono più modificare (nella tabella per la ricerca degli alunni non campere nessuno, neanche digitando nome e cognome

Lascia in bianco i campi delle username, in modo che vengano generate automaticamente. Attualmente, se le imposti in formato diverso da quello previsto crea problemi del tipo indicato.

>     * Inoltre quando si deve aggiungere un alunno devo per forza inserire le informazioni di entrambi i genitori e se l'alunno ne ha solo uno? fare in modo che le informazioni dei genitori o del genitore 2 non siano obbligatorie

Non è necessario inserire i dati dei genitori. Lascia in bianco i dati dei genitori: gli utenti dei genitori verranno creati con dati fittizi, e puoi deciderli di non usarli.

**Comment by trinko** [comment]
> * Possibilità di impostare le valutazioni numeriche quindi scegliere se sono abilitati i voti col + o -, solo col mezzo voto e solo interi. (io per esempio non vorrei che ci fossero i + e i - ma solo voti interi o mezzi)

Ci sono docenti che preferiscono usarli e altri no: perché imporre a tutti i docenti queste limitazioni? O intendi una configurazione valida solo per il proprio utente?

> * Possibilità di scegliere il voto minimo e voto massimo assegnabile

No. La normativa sulle valutazioni è chiara: voto minimo 1 e voto massimo 10.

> * Possibilità di scegliere un numero massimo di voti inseribili per tiplogia per esempio 4+4+4 per periodo scolastico

Perché imporre un numero massimo di valutazioni? Se il docente ritiene di dover fare una valutazione in più, perché impedirglielo?

> * Possibilità da parte dei docenti di inserire il voto in qualsiasi giornata loro vogliano senza il vincolo di aver firmato una lezione in quella giornata per quella materia

No. Il voto deve essere associato ad una lezione svolta. In caso contrario, sarebbe un grave errore formale e potrebbe essere oggetto di ricorso sull'esito finale.

**Comment by Faobi69** [comment]
Capisco, grazie mille comunque, si hai ragione in merito al secondo e terzo punto ma al primo e al secondo no, io aggiungerei delle opzione nella sezione parametri dell'account amministratore in modo che ogni scuola possa decidere in consiglio di istituto quali devono essere le valutazioni utilizzabili (dato che ogni istituto è autonomo) nella mia scuola è in vigore il voto minimo 3 e voto massimo 10 e l'utilizzo dei soli voti interi e voti mezzi senza i + o i -; infatti per provare ho modificato personalmente i file html twig per adattare il registro alle esigenze, però ti chiederi di mettere un'impostazione per evitare di associare un voto a una lezione, se un docente corregge le verifiche di domenica deve per forza inserire il voto nel giorno in cui la verifica è stata svolta?

**Comment by trinko** [comment]
Le segno nella lista delle richieste, ma non so se e quando si potranno fare.
